modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -175,11 +175,30 @@
 
 (defmethod driver/describe-table :bigquery-cloud-sdk
   [_ database {table-name :name, dataset-id :schema}]
-  {:schema dataset-id
-   :name   table-name
-   :fields (-> (.. (get-table database dataset-id table-name) getDefinition getSchema)
-               table-schema->metabase-field-info
-               set)})
+  (let [table                  (get-table database dataset-id table-name)
+        tabledef               (.getDefinition table)
+        ;; a table can only have one partitioned field
+        partitioned-field-name (some-> (or (.getTimePartitioning tabledef)
+                                           (.getRangePartitioning tabledef))
+                                       .getField)
+        fields                 (set
+                                (map
+                                 #(assoc % :database-partitioned (= (:name %) partitioned-field-name))
+                                 (table-schema->metabase-field-info (. tabledef getSchema))))]
+    {:schema dataset-id
+     :name   table-name
+     :fields (cond-> fields
+               ;; if table has time partition but no field is specified as partitioned
+               ;; meaning this table is partitioned by ingestion time
+               ;; so we manually sync this virtual column
+               (and (some? (.getTimePartitioning tabledef))
+                    (nil? partitioned-field-name))
+               (conj
+                {:name                 "_PARTITIONTIME"
+                 :database-type        "TIMESTAMP"
+                 :base-type            :type/DateTimeWithLocalTZ
```

Comment:
Are you sure it shouldn't be `:type/DateTimeWithTZ`? Take a look at `bigquery.tx/base-type->bigquery-type`. Either way I think `driver/describe-table`'s implementation should avoid being coupled to the relationship between `:database-type` and `:base-type` for bigquery.

Reply:
> Are you sure it shouldn't be :type/DateTimeWithTZ

from the types.cljc doc
```clojure
(derive :type/TimeWithLocalTZ :type/TimeWithTZ)    ; a column that is timezone-aware, but normalized to UTC or another offset at rest.
```

And also, this column has the type timestamp, and we are mapping timestamp to `:type/DateTimeWithLocalTZ`, so I'm pretty sure.

Reply:
Oh you're right, I should have been looking at `metabase.driver.bigquery-cloud-sdk/bigquery-type->base-type`, my bad. I think that's a problem with `bigquery.tx/base-type->bigquery-type` then.

Compare:
```clojure
;; metabase.test.data.bigquery-cloud-sdk
(defn base-type->bigquery-type [base-type]
  (let [types {:type/BigInteger     :INTEGER
               :type/Boolean        :BOOLEAN
               :type/Date           :DATE
               :type/DateTime       :DATETIME
               :type/DateTimeWithTZ :TIMESTAMP
               :type/Decimal        :NUMERIC
               :type/Dictionary     :RECORD
               :type/Float          :FLOAT
               :type/Integer        :INTEGER
               :type/Text           :STRING
               :type/Time           :TIME}]
    (or (get types base-type)
        (some base-type->bigquery-type (parents base-type)))))
```
and
```clojure
;; metabase.driver.bigquery-cloud-sdk
(defn- bigquery-type->base-type
  "Returns the base type for the given BigQuery field's `field-mode` and `field-type`. In BQ, an ARRAY of INTEGER has
  \"REPEATED\" as the mode, and \"INTEGER\" as the type name.

  If/when we are able to represent complex types more precisely, we may want to capture that information separately.
  For now, though, we will check if the `field-mode` is \"REPEATED\" and return our :type/Array for that case, then
  proceed to check the `field-type` otherwise."
  [field-mode field-type]
  (if (= Field$Mode/REPEATED field-mode)
    :type/Array
    (case field-type
      "BOOLEAN"    :type/Boolean
      "FLOAT"      :type/Float
      "INTEGER"    :type/Integer
      "RECORD"     :type/Dictionary ; RECORD -> field has a nested schema
      "STRING"     :type/Text
      "DATE"       :type/Date
      "DATETIME"   :type/DateTime
      "TIMESTAMP"  :type/DateTimeWithLocalTZ
      "TIME"       :type/Time
      "NUMERIC"    :type/Decimal
      "BIGNUMERIC" :type/Decimal
      :type/*)))
```

Anyway, my point about coupling still stands. You should use `base-type->bigquery-type` instead of hard-coding `:type/DateTimeWithLocalTZ`, like is done in `table-schema->metabase-field-info`.

Reply:
yes I did that in https://github.com/metabase/metabase/pull/36892/commits/4c19a94e14213e4b972416f02aed15455480e8ea